ReactApp/src/utils/rest-api-call.js

```diff
@@ -105,20 +105,23 @@ async function extractDataFromResponse(response){
     return retObject
 }
 
+
 export async function requestServer(path, method, body) {
     var response
+    let responsePromise
+    let refreshTokenPromise = undefined
     if (Cookies.get(TOKEN_FIELD_NAME)) {
-        response = await fetchWithCookies(path,method,body)
+      responsePromise = fetchWithCookies(path,method,body)
 
-        // refresh cookies
-         fetchWithCookies(serverPath.REFRESH_TOKEN ,httpsMethod.GET, undefined).then(async (refreshTokenResponse)=>{
+      // refresh cookies
+      refreshTokenPromise = fetchWithCookies(serverPath.REFRESH_TOKEN ,httpsMethod.GET, undefined).then(async (refreshTokenResponse)=>{
```

Comment:
why does the callback passed to the `then` has to be `async`?

Reply:
because extractDataFromResponse is async function.

I need to make sure I assign the token before exiting requestServer(), for that I have to wait the token is extract.